test/java/seedu/addressbook/logic/LogicTest.java

```diff
@@ -90,7 +91,13 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+
+
+        Command command  = new Parser().parseCommand(inputCommand);
+        if(command.isMutating()){
+            assertEquals(addressBook,saveFile.load());
+        }
```

Comment:
This conditional statement tests for the case where the command is a mutating command. What about the case where the command is not supposed to mutate? In those cases, we should test as to whether the address book was modified.